src/ape/api/accounts.py

```diff
@@ -93,10 +93,16 @@ def call(self, txn: TransactionAPI, send_everything: bool = False) -> ReceiptAPI
         txn = self.prepare_transaction(txn)
 
         if send_everything:
+            error_message = "Sender does not have enough to cover transaction value and gas: "
             if txn.max_fee is None:
                 raise TransactionError(message="Max fee must not be None.")
-
-            txn.value = self.balance - txn.max_fee
+            if txn.gas_limit:
+                txn.value = self.balance - (txn.max_fee * txn.gas_limit)
```

Comment:
I think there is a property on txn for total cost or something

Reply:
I don't see that property: 
```
(Pdb) 'total_cost' in dir(txn)
False
```

Reply:
once the total cost is calculated correctly I should be able to use that as to `assert sender.balance==0` in the `send_everything=True` test